### PR TITLE
#17478 OIDC add offline_access

### PR DESCRIPTION
**Pull request by spark-c**
This PR updates the OIDC documentation pages to include the `offline_access` scope and show how to enable the use of refresh tokens on the various OIDC platforms. Relevant screenshots and videos have been replaced.

Since Github sees all of this content is brand new, I've added comments throughout the affected documents where changes have been made; they look like this:
```
<!-- updated for 17478 -->
```
 You should be able to find all of the necessary updated sections by searching for the ticket number `17478`.

[Original ticket](https://github.com/appsmithorg/appsmith/issues/17478)

### CODE REVIEW

File: .github/drafts/17478/17478_Active_Directory_Draft.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+1. Go to API permissions on the left panel.
+2. Click on Add a permission button.
+3. Under Request API permissions, Select Microsoft Graph and click on Delegated permissions.
+4. Select the following permissions:
+   1. Under **OpenID** permissions-
+      1. email
+      2. offline_access
+      3. openid
+      4. Profile
```
Comment (nsarupr) [inline .github/drafts/17478/17478_Active_Directory_Draft.md:86]: this should be `profile`

File: .github/drafts/17478/17478_Active_Directory_Draft.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+   2. Under **User** permissions-
+      1. User.Read
+5. Once complete, these scopes will get added to the Active Directory application.
+6. Click on **Grant admin consent for Default Directory** on the top of the table, next to Add a permission button.
+
+### Configure Username Attributes for Active Directory
+
+The username attributes define the attributes used as usernames for authentication. You can add the attribute to this field that you consider for logging.
+
+![Appsmith Username Attribute](<../../../../../.gitbook/assets/OIDC \_ Active Directory \_ Username Attribute \_ sub.png>)
```
Comment (nsarupr) [inline .github/drafts/17478/17478_Active_Directory_Draft.md:96]: The image here doesn't have `offline_access` in the scope which might confuse users.

File: .github/drafts/17478/17478_Okta_Draft.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+To continue with the OIDC setup on Appsmith, navigate to the fields on the Okta configurations, and perform the actions as mentioned below:
+
+*   On the Okta Application homepage, go to the **General** tab:
+
+    * Copy the **Client ID** and **Client Secret** and paste them into the **OIDC** configurations in Appsmith.
+
+
+
+![Client ID, Client Secret, and Okta Domain](<../../../../../.gitbook/assets/Screenshot 2022-07-18 at 11.45.21 AM.png>)
```
Comment (nsarupr) [inline .github/drafts/17478/17478_Okta_Draft.md:50]: Can we please rename the image.

File: .github/drafts/17478/17478_Ping_Identity_Draft.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+* In the Attribute Mapping, add the OIDC attributes. Please note that only the values added in the `Scope Grant` will be valid here. Hit **Save and Close**.
+
+{% embed url="https://youtu.be/fRreXB6P0No" %}
+Configure Environment and Scope
+{% endembed %}
+
+<!-- added for 17478 -->
+* In your app's settings pane, navigate to the **Configuration** tab. Click the pencil icon to begin editing the app's settings, and tick the **Refresh Token** box to enable Ping Identity to issue refresh tokens to Appsmith. Your app is now configured for use with Appsmith.
+
+{% embed url="https://youtu.be/y7f2kXY9zcw" %} <!-- added for 17478 -->
```
Comment (nsarupr) [inline .github/drafts/17478/17478_Ping_Identity_Draft.md:45]: Kudos!!!
Better than an image.

Comment (spark-c) [reply .github/drafts/17478/17478_Ping_Identity_Draft.md:45]: Thanks!

File: .github/drafts/17478/17478_Okta_Draft.md (hunk 3 above)
Comment (spark-c) [reply .github/drafts/17478/17478_Okta_Draft.md:50]: Updated the image caption to: "Use these Okta values to configure OIDC on Appsmith"

Let me know if you have a different suggestion!

**Review by trishaanand — APPROVED** [review]

**Pull request merged**